src/TestFramework/Codeception/Adapter/CodeceptionAdapter.php

```diff
@@ -0,0 +1,395 @@
+
+        $process = new Process($testFrameworkVersionExecutable);
+        $process->mustRun();
+
+        $version = 'unknown';
+
+        try {
+            $version = $this->versionParser->parse($process->getOutput());
+        } catch (\InvalidArgumentException $e) {
+            $version = 'unknown';
+        } finally {
+            $this->cachedVersion = $version;
+        }
+
+        return $this->cachedVersion;
+    }
+
+    public function getInitialTestsFailRecommendations(string $commandLine): string
+    {
+        return sprintf('Check the executed command to identify the problem: %s', $commandLine);
+    }
+
+    protected function getInterceptorFileContent(string $interceptorPath, string $originalFilePath, string $mutatedFilePath): string
+    {
+        $infectionPhar = '';
+
+        if (0 === strpos(__FILE__, 'phar:')) {
+            $infectionPhar = sprintf(
+                '\Phar::loadPhar("%s", "%s");',
+                str_replace('phar://', '', \Phar::running(true)),
+                'infection.phar'
+            );
+        }
+
+        $namespacePrefix = $this->getInterceptorNamespacePrefix();
+
+        return <<<CONTENT
+{$infectionPhar}
+require_once '{$interceptorPath}';
+
+use {$namespacePrefix}Infection\StreamWrapper\IncludeInterceptor;
+
+IncludeInterceptor::intercept('{$originalFilePath}', '{$mutatedFilePath}');
+IncludeInterceptor::enable();
+CONTENT;
+    }
+
+    private function createCustomBootstrapWithInterceptor(MutantInterface $mutant): string
+    {
+        $originalFilePath = $mutant->getMutation()->getOriginalFilePath();
+        $mutatedFilePath = $mutant->getMutatedFilePath();
+
+        $originalBootstrap = $this->getOriginalBootstrapFilePath();
+        $bootstrapPlaceholder = $originalBootstrap ? "require_once '{$originalBootstrap}';" : '';
+
+        $interceptorPath = \dirname(__DIR__, 3) . '/StreamWrapper/IncludeInterceptor.php';
+
+        $customBootstrap = <<<AUTOLOAD
+<?php
+
+%s
+%s
+
+AUTOLOAD;
+
+        return sprintf(
+            $customBootstrap,
+            $bootstrapPlaceholder,
+            $this->getInterceptorFileContent($interceptorPath, $originalFilePath, $mutatedFilePath)
+        );
+    }
+
+    private function getOriginalBootstrapFilePath(): ?string
+    {
+        if (!\array_key_exists('bootstrap', $this->originalConfigContentParsed)) {
+            return null;
+        }
+
+        if ($this->filesystem->isAbsolutePath($this->originalConfigContentParsed['bootstrap'])) {
+            return $this->originalConfigContentParsed['bootstrap'];
+        }
+
+        return sprintf(
+            '%s/%s/%s',
+            $this->projectDir,
+            $this->originalConfigContentParsed['paths']['tests'] ?? 'tests',
+            $this->originalConfigContentParsed['bootstrap']
+        );
+    }
+
+    private function getInterceptorNamespacePrefix(): string
+    {
+        $prefix = strstr(__NAMESPACE__, 'Infection', true);
+        \assert(\is_string($prefix));
```

Comment:
I'm a bit confused, why the `assert()` here instead of `Assert::`?

Reply:
exactly